include/openthread/tcat.h

```diff
@@ -0,0 +1,173 @@
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file
+ * @brief
+ *  This file defines the top-level functions for the OpenThread TCAT.
+ *
+ *  @note
+ *   The functions in this module require the build-time feature `OPENTHREAD_CONFIG_BLE_TCAT_ENABLE=1`.
+ *
+ *  @note
+ *   To enable cipher suite DTLS_PSK_WITH_AES_128_CCM_8, MBEDTLS_KEY_EXCHANGE_PSK_ENABLED
+ *    must be enabled in mbedtls-config.h
+ *   To enable cipher suite DTLS_ECDHE_ECDSA_WITH_AES_128_CCM_8,
+ *    MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED must be enabled in mbedtls-config.h.
+ */
+
+#ifndef OPENTHREAD_TCAT_H_
+#define OPENTHREAD_TCAT_H_
+
+#include <stdint.h>
+#include <openthread/message.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @addtogroup api-ble-secure
+ *
+ * @brief
+ *   This module includes functions that implement TCAT communication.
+ *
+ *   The functions in this module are available when TCAT feature
+ *   (`OPENTHREAD_CONFIG_BLE_TCAT_ENABLE`) is enabled.
+ *
+ * @{
+ *
+ */
+
+#define OT_TCAT_MAX_SERVICE_NAME_LENGTH \
+    15 ///< Maximum string length of a UDP or TCP service name (does not include null char).
+
+/**
+ * Represents TCAT status code.
+ *
+ */
+typedef enum otTcatStatusCode
+{
+    OT_TCAT_STATUS_SUCCESS       = 0, ///< Command or request was successfully processed
+    OT_TCAT_STATUS_UNSUPPORTED   = 1, ///< Requested command or received TLV is not supported
+    OT_TCAT_STATUS_PARSE_ERROR   = 2, ///< Request / command could not be parsed correctly
+    OT_TCAT_STATUS_VALUE_ERROR   = 3, ///< The value of the transmitted TLV has an error
+    OT_TCAT_STATUS_GENERAL_ERROR = 4, ///< An error not matching any other category occurred
+    OT_TCAT_STATUS_BUSY          = 5, ///< Command cannot be executed because the resource is busy
+    OT_TCAT_STATUS_UNDEFINED    = 6, ///< The requested value, data or service is not defined (currently) or not present
+    OT_TCAT_STATUS_HASH_ERROR   = 7, ///< The hash value presented by the commissioner was incorrect
+    OT_TCAT_STATUS_UNAUTHORIZED = 8, ///< Sender does not have sufficient authorization for the given command
```

Comment:
Values 0x10-0x17 represent 'Unauthorized' (see latest draft4 spec). From perspective of the Thread device, only value 0x10 is relevant today as the other bit patterns haven't been defined yet. So:

```suggestion
    OT_TCAT_STATUS_UNAUTHORIZED = 0x10, ///< Sender does not have sufficient authorization for the given command
```

Reply:
fixed.